src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -116,42 +121,59 @@ protected void runTool(final JavaSparkContext ctx) {
         //TOOO: should this use getUnfilteredReads? getReads will apply default and command line filters
         final JavaRDD<GATKRead> initialReads = getReads();
 
-        final JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder(), getRecommendedNumReducers());
+        final JavaRDD<GATKRead> alignedReads;
+        final SAMFileHeader header;
+        final BwaSparkEngine engine;
+        if (align) {
+                engine = new BwaSparkEngine(ctx, referenceArguments.getReferenceFileName(), bwaArgs.indexImageFile, getHeaderForReads(), getReferenceSequenceDictionary());
+                alignedReads = !bwaArgs.singleEndAlignment ? engine.alignPaired(getReads()) : engine.alignUnpaired(getReads());
```

Comment:
You're calling `getReads()` again here -- can you just use `initialReads` above? Is this expensive?

Reply:
Also, `getReads()` does filtering -- did you want to pass in the unfiltered reads (`getUnfilteredReads()`) to BWA?

Reply:
1.  It doesn't make a difference, since it is lazy. But I've changed the code to use `initialReads` to make it less confusing.
2. Good question! What do you think is best, @droazen?

Reply:
I would think that you'd want to pass BWA the unfiltered reads, and then do filtering after alignment. Do you agree, @tedsharpe ?

Reply:
On the one hand, it seems wasteful to spend time aligning reads that you're about to discard.  On the other hand, if you're doing pairwise alignment and your filters don't either pass or reject *both* members of the pair consistently, you'll have a royal mess.  Since filters are designed to operate on reads, not on read pairs, it would be hard (but not impossible) to guarantee correct filter behavior in this context.
All of which is a long-winded way of saying that you probably want to align unfiltered reads like @droazen said.

Reply:
Thanks for the comments @tedsharpe and @droazen. I've updated the code to align on unfiltered reads then do the alignment. I've updated `BwaSpark` and `BwaAndMarkDuplicatesPipelineSpark` too. (BTW do we still need the latter do you think?)